pkg/storage/replica.go

```diff
@@ -3275,6 +3275,13 @@ func (r *Replica) processRaftCommand(
 			}
 			return errors.Errorf("lease %s not held", l)
 		}
+		if raftCmd.BatchRequest.IsSingleSkipLeaseCheckRequest() {
+			// For lease commands, no additional verification is needed if the origin
+			// lease isn't valid.
+			if !r.isLeaseValidLocked(raftCmd.OriginLease, ts) {
```

Comment:
What prompted this change? It's definitely not correct however, as the results of `r.isLeaseValidLocked` can be different between different nodes, as node liveness information which is gossiped can easily be different. In other words, one replica might see an incremented epoch and decide there's no point in checking the lease and the other might not see it and go ahead and check the lease. Then you'd end up with a consistency error and perhaps worse, potentially two replicas which believe they have the lease.